apps/accounts/admin.py

```diff
@@ -26,3 +28,36 @@ class UserAdmin(BaseUserAdmin):
         ('Organisations', {'fields': ('organisations',)}),
         ('Important dates', {'fields': ('last_login', 'date_joined'), 'classes': ('collapse',)}),
     ]
+
+    def response_change(self, request, obj):
+        """ Add permission to the users if it's set to staff and belongs to the organisation. """
+        response = super().response_change(request, obj)
+        self.add_remove_permissions(obj)
+        return response
+
+    def get_queryset(self, request):
```

Comment:
This throws an error (added below), here is how to replicate it:
1) Add organisation 'One'
2) Add organisation 'Two'
3) Add user test@dev.ngo
4) Set test user as staff but NOT superuser and add organisation 'One'
5) Login as test
6) Edit the test user (using the test account)
7) Add organisation 'Two'
8) Save and upon returning you see the user is duplicated in the list view
9) Editing the test user again results in an exception
```

Environment:


Request Method: GET
Request URL: http://localhost:8000/admin/accounts/user/2/change/

Django Version: 1.11.10
Python Version: 3.5.2
Installed Applications:
['blanc_admin_theme',
 'core',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.sites',
 'ckeditor',
 'raven.contrib.django.raven_compat',
 'directory.apps.DirectoryConfig',
 'accounts.apps.AccountsConfig',
 'django_extensions',
 'debug_toolbar']
Installed Middleware:
['raven.contrib.django.middleware.DjangoRestFrameworkCompatMiddleware',
 'raven.contrib.django.middleware.SentryMiddleware',
 'debug_toolbar.middleware.DebugToolbarMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'django.contrib.sites.middleware.CurrentSiteMiddleware']


Traceback:

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/core/handlers/exception.py" in inner
  41.             response = get_response(request)

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/contrib/admin/options.py" in wrapper
  551.                 return self.admin_site.admin_view(view)(*args, **kwargs)

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/utils/decorators.py" in _wrapped_view
  149.                     response = view_func(request, *args, **kwargs)

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/views/decorators/cache.py" in _wrapped_view_func
  57.         response = view_func(request, *args, **kwargs)

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/contrib/admin/sites.py" in inner
  224.             return view(request, *args, **kwargs)

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/contrib/admin/options.py" in change_view
  1511.         return self.changeform_view(request, object_id, form_url, extra_context)

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/utils/decorators.py" in _wrapper
  67.             return bound_func(*args, **kwargs)

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/utils/decorators.py" in _wrapped_view
  149.                     response = view_func(request, *args, **kwargs)

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/utils/decorators.py" in bound_func
  63.                 return func.__get__(self, type(self))(*args2, **kwargs2)

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/contrib/admin/options.py" in changeform_view
  1408.             return self._changeform_view(request, object_id, form_url, extra_context)

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/contrib/admin/options.py" in _changeform_view
  1429.             obj = self.get_object(request, unquote(object_id), to_field)

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/contrib/admin/options.py" in get_object
  672.             return queryset.get(**{field.name: object_id})

File "/home/martin/.virtualenvs/commonslibrary/lib/python3.5/site-packages/django/db/models/query.py" in get
  384.             (self.model._meta.object_name, num)

Exception Type: MultipleObjectsReturned at /admin/accounts/user/2/change/
Exception Value: get() returned more than one User -- it returned 2!
```